test/e2e/operator_test.go

```diff
@@ -0,0 +1,82 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	// kclient is the Kubernetes client used for e2e tests.
+	kclient client.Client
+	// ctx is an empty context used for client calls.
+	ctx = context.TODO()
+	// operatorName is the name of the operator.
+	operatorName = "contour-operator"
+	// operatorNs is the name of the operator's namespace.
+	operatorNs = "contour-operator"
+	// defaultContourNs is the default spec.namespace.name of a Contour.
+	defaultContourNs = "projectcontour"
+	// testUrl is the url used to test e2e functionality.
+	testUrl = "http://local.projectcontour.io/"
+	// expectedDeploymentConditions are the expected status conditions of a
+	// deployment.
+	expectedDeploymentConditions = []appsv1.DeploymentCondition{
+		{Type: appsv1.DeploymentAvailable, Status: corev1.ConditionTrue},
+	}
+)
+
+func TestMain(m *testing.M) {
+	cl, err := newClient()
+	if err != nil {
+		os.Exit(1)
+	}
+	kclient = cl
+
+	os.Exit(m.Run())
+}
+
+func TestOperatorDeploymentAvailable(t *testing.T) {
+	t.Helper()
+	if err := waitForDeploymentStatusConditions(t, ctx, kclient, 3*time.Minute, operatorName, operatorNs, expectedDeploymentConditions...); err != nil {
+		t.Errorf("failed to observe expected conditions for deployment %s/%s: %v", operatorNs, operatorName, err)
+	}
+}
+
+func TestDefaultContour(t *testing.T) {
+	testName := "test-default-contour"
+	newContour(t, ctx, kclient, testName, operatorNs)
+	defer deleteContour(t, ctx, kclient, 3*time.Minute, testName, operatorNs)
+	// Defer the namespace deletion since contour.spec.namespace.removeOnDeletion
+	// defaults to false.
+	defer deleteNamespace(t, ctx, kclient, 3*time.Minute, defaultContourNs)
+	expectedContourConditions := []metav1.Condition{
+		{Type: operatorv1alpha1.ContourAvailableConditionType, Status: metav1.ConditionTrue},
+		// TODO [danehans]: Update when additional status conditions are added to Contour.
+	}
+	if err := waitForContourStatusConditions(t, ctx, kclient, 5*time.Minute, testName, operatorNs, expectedContourConditions...); err != nil {
+		getContourStatus(t, ctx, kclient, testName, operatorNs)
+		listPodsInNs(t, ctx, kclient, operatorNs)
+		t.Errorf("failed to observe expected conditions for contour %s/%s: %v", operatorNs, testName, err)
+	}
+	t.Logf("observed available status condition for contour %s/%s", testName, operatorNs)
+
+	appName := "kuard"
+	newDeployment(t, ctx, kclient, appName, defaultContourNs, "gcr.io/kuar-demo/kuard-amd64:1", 3)
+	if err := waitForDeploymentStatusConditions(t, ctx, kclient, 3*time.Minute, appName, defaultContourNs, expectedDeploymentConditions...); err != nil {
+		t.Errorf("failed to observe expected conditions for deployment %s/%s: %v", defaultContourNs, appName, err)
+	}
+	newClusterIPService(t, ctx, kclient, appName, defaultContourNs, 80, 8080)
+	newIngress(t, ctx, kclient, appName, defaultContourNs, appName, 80)
+	waitForCurl(t, testUrl, 1*time.Minute)
```

Comment:
In general, I think it's a better approach to write utilities to return errors, then simply use `assert.NoError` in the test caller. That ends up generating better error context in most cases. Not a blocker for this review, but maybe plan to revisit in the future.